src/ImageSharp/Color/Color.Conversions.cs

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System.Numerics;
+
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace SixLabors.ImageSharp
+{
+    /// <content>
+    /// Contains constructors and implicit conversion methods.
+    /// </content>
+    public readonly partial struct Color
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Color"/> struct.
+        /// </summary>
+        /// <param name="pixel">The <see cref="Rgba64"/> containing the color information.</param>
+        public Color(Rgba64 pixel)
+        {
+            this.data = pixel;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Color"/> struct.
+        /// </summary>
+        /// <param name="pixel">The <see cref="Rgba32"/> containing the color information.</param>
+        public Color(Rgba32 pixel)
+        {
+            this.data = new Rgba64(pixel);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Color"/> struct.
+        /// </summary>
+        /// <param name="pixel">The <see cref="Argb32"/> containing the color information.</param>
+        public Color(Argb32 pixel)
+        {
+            this.data = new Rgba64(pixel);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Color"/> struct.
+        /// </summary>
+        /// <param name="pixel">The <see cref="Bgra32"/> containing the color information.</param>
+        public Color(Bgra32 pixel)
+        {
+            this.data = new Rgba64(pixel);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Color"/> struct.
+        /// </summary>
+        /// <param name="pixel">The <see cref="Rgb24"/> containing the color information.</param>
+        public Color(Rgb24 pixel)
+        {
+            this.data = new Rgba64(pixel);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Color"/> struct.
+        /// </summary>
+        /// <param name="pixel">The <see cref="Bgr24"/> containing the color information.</param>
+        public Color(Bgr24 pixel)
+        {
+            this.data = new Rgba64(pixel);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Color"/> struct.
+        /// </summary>
+        /// <param name="vector">The <see cref="Vector4"/> containing the color information.</param>
+        public Color(Vector4 vector)
+        {
+            this.data = new Rgba64(vector);
+        }
+
+        /// <summary>
+        /// Converts an <see cref="Rgba64"/> to <see cref="Color"/>.
+        /// </summary>
+        /// <param name="source">The <see cref="Rgba64"/>.</param>
+        /// <returns>The <see cref="Color"/>.</returns>
+        public static implicit operator Color(Rgba64 source) => new Color(source);
```

Comment:
shouldn't these implicit operators be on the Pixel type instead of Color? there are alot more Pixel types after all.

Reply:
Good point! I think I will change this.